### PR TITLE
feat: Added /getPrintJob and fixed /query

### PR DESCRIPTION
**Pull request by Caleb-Biggs**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,25 @@
...
+MongoDB shell cheat sheet
+https://www.slingacademy.com/article/mongodb-shell-commands-the-complete-cheat-sheet/#Advanced_Querying
+
 <h5>Database API sample curl requests</h5>
+<h6>Sample GET request</h6>
+
+These two curl requests do the same thing:
+curl -X GET "http://54.200.253.84:80/query?CollectionName=PrintJob&Query=%7B%22Title%22%3A%22PrintJob%201%22%7D"
+
+curl -G -X GET http://54.200.253.84:80/query --data-urlencode "CollectionName=PrintJob" --data-urlencode "Query={\"Title\":\"PrintJob 1\"}"
```
Comment (amychisholm03) [inline README.md]: Could you replace all instances of the IP to the domain name (`http://api.wsuv-hp-capstone.com/`)?

File: server.js [hunk 2]
```diff
@@ -50,6 +50,28 @@ function setupGets(database) {
...
 
 
+  fastify.get('/query', async (request, reply) => {
+    let message = "";
+    let code = 200;
+    const Query = JSON.parse(request.query.Query);
+    const Collection = database.collection(request.query.CollectionName);
+    try { message = await Collection.find(Query).toArray(); }
+    catch(err){ message = err; code = 500; }
+    reply.code(code).send(message);
```
Comment (amychisholm03) [inline server.js:60]: Maybe have this (line 60) in a `finally` block?

File: test/integration/server.test.js [hunk 3]
```diff
@@ -63,17 +63,25 @@ test('POST /createWorkflowStep', async (t) => {
...
-        url: '/query',
-        headers: {
-            'Content-Type': 'application/json'
-        },
-        body: '{"CollectionName": "PrintJob", "Query": {"Title": "PrintJob 1"}}'
+        method: 'GET',
+        url:`/query?CollectionName=${CollectionName}&Query=${Query}`
+    });
+    assert.strictEqual(response.statusCode, 200);
+    console.log("Query Results:", JSON.parse(response.payload));
```
Comment (amychisholm03) [inline test/integration/server.test.js:74]: At least assert `response.payload` contains something

File: server.js [hunk 4]
```diff
@@ -50,6 +50,28 @@ function setupGets(database) {
...
+  });
+
+
+  fastify.get('/getPrintJob', async (request, reply) => {
+    let message = "";
+    let code = 200;
+    const PrintJob = request.query.PrintJobName;
```
Comment (amychisholm03) [inline server.js]: Why `PrintJobName` instead of `Title` like the other APIs?

File: server.js [hunk 5]
```diff
@@ -126,14 +139,10 @@ async function fastifyPostHelper(reply, database, func, args) {
...
-function main(){
-  start();
-}
-
-
-// This is needed so that server.test.js doesn't run main()
+// This allows passing in an alternate port as a command line argument
 if (require.main === module) {
-  main();
+  if(process.argv.length > 2) start("0.0.0.0", process.argv[2]);
```
Comment (amychisholm03) [inline server.js:152]: Ah nice touch

**Review by amychisholm03 — CHANGES_REQUESTED** [review]

**Review by amychisholm03 — DISMISSED** [review]

**Review by CadeTormanen — APPROVED** [review]
looks great

**Review by amychisholm03 — APPROVED** [review]
ship it

**Pull request merged**